Remora.Discord.Commands/Extensions/CommandTreeExtensions.cs

```diff
@@ -479,13 +480,33 @@ ILocalizationProvider localizationProvider
                 );
             }
 
-            if (command.CommandMethod.GetParameters().Length > 0)
+            var parameters = command.CommandMethod.GetParameters();
+            if (parameters.Length > 0)
             {
-                return new UnsupportedFeatureError
-                (
-                    "Context menu commands may not have parameters.",
-                    command
-                );
+                var expectedParameter = commandType switch
+                {
+                    ApplicationCommandType.Message => "message",
+                    ApplicationCommandType.User => "user",
+                    _ => null
+                };
+
+                if (expectedParameter == null)
+                {
+                    return new UnsupportedFeatureError
+                    (
+                        $"Command type {commandType} is not supported.",
+                        command
+                    );
+                }
+
+                if (parameters.Length != 1 || parameters[0].Name != expectedParameter)
+                {
+                    return new UnsupportedFeatureError
```

Comment:
This is not an unsupported feature; rather, it is a programmer error. Use a more appropriate error type.

Reply:
I mean we could actually throw here as it would be a developer error.